app/views/components/input_date_field.scala.html

```diff
@@ -0,0 +1,57 @@
+@*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *@
+
+@(input: models.ui.DateInput, name: String, form: Form[_])(implicit request: Request[_], messages: Messages, ctx: PageContext)
+
+@optionalAria() = @{
+  Seq(input.hint.map(_ => s"$name-item-hint"), input.errorMsgs.headOption.map(err => s"${err.id}-error")).flatten match {
+    case Nil => ""
+    case l => ariaDescribedBy("input-hint")
+  }
+}
+<div class="govuk-date-input" id="date-input">
```

Comment:
The id here is not going to be unique if there are multiple dates on a form. Further, if there are multiple components on a page a label is created for the id "name-0", should this be the id?

Reply:
As this assumes one input per page, adding -0 at end of id doesn't really make much sense.  The date fields are called day, month and year, so would have to change if we have multiple dates on a page - but this woud be the least of the changes compared to controller side of things.

Reply:
Ok